tools/crm_diff.c

```diff
@@ -19,44 +19,103 @@
 
 #include <crm/crm.h>
 #include <crm/msg_xml.h>
+#include <crm/common/cmdline_internal.h>
 #include <crm/common/xml.h>
 #include <crm/common/ipc.h>
 #include <crm/cib.h>
 
-/* *INDENT-OFF* */
-static struct crm_option long_options[] = {
-    /* Top-level Options */
-    {"help",           0, 0, '?', "\t\tThis text"},
-    {"version",        0, 0, '$', "\t\tVersion information"  },
-    {"verbose",        0, 0, 'V', "\t\tIncrease debug output\n"},
-
-    {"-spacer-",	1, 0, '-', "\nOriginal XML:"},
-    {"original",	1, 0, 'o', "\tXML is contained in the named file"},
-    {"original-string", 1, 0, 'O', "XML is contained in the supplied string"},
-
-    {"-spacer-",	1, 0, '-', "\nOperation:"},
-    {"new",		1, 0, 'n', "\tCompare the original XML to the contents of the named file"},
-    {"new-string",      1, 0, 'N', "\tCompare the original XML to the contents of the supplied string"},
-    {"patch",		1, 0, 'p', "\tPatch the original XML with the contents of the named file"},
-
-    {"-spacer-", 1, 0, '-', "\nAdditional Options:"},
-    {"cib",	 0, 0, 'c', "\t\tCompare/patch the inputs as a CIB (includes versions details)"},
-    {"stdin",	 0, 0, 's', NULL, 1},
-    {"no-version", 0, 0, 'u', "\tGenerate the difference without versions details"},
-    {"-spacer-", 1, 0, '-', "\nExamples:", pcmk_option_paragraph},
-    {"-spacer-", 1, 0, '-', "Obtain the two different configuration files by running cibadmin on the two cluster setups to compare:", pcmk_option_paragraph},
-    {"-spacer-", 1, 0, '-', " cibadmin --query > cib-old.xml", pcmk_option_example},
-    {"-spacer-", 1, 0, '-', " cibadmin --query > cib-new.xml", pcmk_option_example},
-    {"-spacer-", 1, 0, '-', "Calculate and save the difference between the two files:", pcmk_option_paragraph},
-    {"-spacer-", 1, 0, '-', " crm_diff --original cib-old.xml --new cib-new.xml > patch.xml", pcmk_option_example },
-    {"-spacer-", 1, 0, '-', "Apply the patch to the original file:", pcmk_option_paragraph },
-    {"-spacer-", 1, 0, '-', " crm_diff --original cib-old.xml --patch patch.xml > updated.xml", pcmk_option_example },
-    {"-spacer-", 1, 0, '-', "Apply the patch to the running cluster:", pcmk_option_paragraph },
-    {"-spacer-", 1, 0, '-', " cibadmin --patch patch.xml", pcmk_option_example },
-
-    {0, 0, 0, 0}
+#define SUMMARY "Compare two Pacemaker configurations (in XML format) to produce a custom diff-like output, " \
+                "or apply such an output as a patch"
+
+struct {
+    gboolean apply;
+    gboolean as_cib;
+    gboolean no_version;
+    gboolean raw_1;
+    gboolean raw_2;
+    gboolean use_stdin;
+    char *xml_file_1;
+    char *xml_file_2;
+} options;
```

Comment:
I think you either have to make this static or init it to { FALSE, } to ensure it's all zeroes to start with

Reply:
Never mind, globals are zeroed